tests/4.5/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_if_no_modifier.F90

```diff
@@ -0,0 +1,141 @@
+! We create three tests, one for no directive, another one for the target 
+! directive and another one for the parallel directive.
+!
+!===--------------------------------------------------------------------------===!  
+
+#include "ompvv.F90"
+
+#define ATTEMPT_THRESHOLD 70
+#define NUM_ATTEMPTS 100
+#define N 1024
+
+PROGRAM test_target_teams_distribute_parallel_for_if_no_modifier
+   USE iso_fortran_env
+   USE ompvv_lib
+   USE omp_lib
+   implicit none
+
+   OMPVV_TEST_VERBOSE(target_teams_distribute_if_no_modifier() .ne. 0)
+   OMPVV_REPORT_AND_RETURN()
+
+
+CONTAINS 
+   LOGICAL FUNCTION checkPreconditions()
+      LOGICAL :: isOffloading
+      INTEGER :: i, raiseWarningDevice, raiseWarningHost
+      INTEGER, DIMENSION(N) :: init_num_threads_dev, init_num_threads_host
+      ! Test for offloading
+      isOffloading = .FALSE.
+      OMPVV_TEST_AND_SET_OFFLOADING(isOffloading)
+      OMPVV_WARNING_IF(isOffloading .eqv. .FALSE. , "With offloading off, it is not&
+                  &possible to test if code executes on parallel or target")
+  
+      ! Initialize init_num_threads_dev and init_num_threads_host arrays
+      DO i = 1, N
+         init_num_threads_dev(i) = 0
+         init_num_threads_host(i) = 0
+      END DO
+
+      ! Get the init_num_threads for host and device, if it is 1 then parallel
+      ! cannot be tested
+      !$omp target teams distribute parallel do num_threads(OMPVV_NUM_THREADS_DEVICE)
+      DO i = 1, N
+         init_num_threads_dev(i) = omp_get_num_threads()
+      END DO
+   
+      !$omp parallel do num_threads(OMPVV_NUM_THREADS_DEVICE)
+      DO i = 1, N 
+         init_num_threads_host(i) = omp_get_num_threads()
+      END DO
+
+      raiseWarningDevice = 0
+      raiseWarningHost = 0
+      
+      DO i = 1, N
+         IF (init_num_threads_dev(i) .gt. 1) THEN
+            raiseWarningDevice = raiseWarningDevice + 1
+         END IF
+         IF (init_num_threads_host(i) > 1) THEN
+            raiseWarningHost = raiseWarningHost + 1
+         END IF
+      END DO
+
+      OMPVV_WARNING_IF(raiseWarningDevice .eq. 0, "Initial number of threads on&
+      & device was 1. It is not possible to test the if for the parallel directive")
+ 
+      OMPVV_WARNING_IF(raiseWarningHost .eq. 0, "Initial number of threads on&
+      & host was 1. It is not possible to test the if for the parallel directive")
+ 
+   checkPreconditions = isOffloading
+   END FUNCTION checkPreconditions
+
+   INTEGER FUNCTION target_teams_distribute_if_no_modifier()
+      INTEGER, DIMENSION(N) :: a, warning 
+      LOGICAL :: isOffloading  
+      INTEGER :: errors, attempt, i, raiseWarning 
+      errors = 0 
+      attempt = 0
+
+      isOffloading = checkPreconditions()  
+         
+      DO i = 1, N
+         a(i) = 1
+         warning(i) = 0
+      END DO
+ 
+      DO attempt = 1, NUM_ATTEMPTS
+         !$omp target teams distribute parallel do if(attempt .ge. ATTEMPT_THRESHOLD) &
```

Comment:
Simplify, if clause is true: ensure it is running on device, produce error otherwise. Still map and modify a

Reply:
It looks OK. 
One very minor issue is that the behavior of the Fortran version is different from that of the C version.
The C version offloads the compute region NUM_ATTEMPS - ATTEMPT_THRESHOLD times out of NUM_ATTMPTS trials, but the Fortran version offloads the compute region  NUM_ATTEMPS - ATTEMPT_THRESHOLD times out of NUM_ATTMPTS - 1 trials. (The Fortran version executes the compute region serially on the host one less times than the C version.)
One way to make the Fortran version behave exactly same as the C version would be to change the if condition in line 100 to "attempt .gt. ATTEMPT_THRESHOLD" and revert "NUM_ATTEMPTS - 1" in line 99 back to "NUM_ATTEMPTS".

Reply:
@seyonglee thank you for this! Making the change now